### PR TITLE
Add ability to change beacon URL. Adds script to the head

### PR DESCRIPTION
**Pull request by Ilya-Hubich**
Changes:
* Add ability to change beacon URL. 
* Adds the newRleic script to the head according to the recommendations: https://docs.newrelic.com/docs/browser/single-page-app-monitoring/get-started/install-single-page-app-monitoring

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ class HtmlWebpackNewRelicPlugin {
         applicationID: undefined,
         license: undefined,
         trustKey: undefined,
+        beacon: undefined,
```
Comment (robrap) [inline src/index.js:13]: Wouldn't the default need to be `bam-cell.nr-data.net` for backward-compatibility?

File: package.json [hunk 2]
```diff
@@ -12,8 +12,7 @@
     "test:examples": "node examples/",
     "cover": "istanbul cover _mocha -- --require babel-core/register --require babel-polyfill --recursive",
     "lint": "eslint src test",
-    "build": "cross-env BABEL_ENV=production babel src --out-dir lib",
-    "prepare": "npm run clean && npm run lint && npm run test && npm run build"
```
Comment (robrap) [inline package.json:16]: Why is this being removed?

File: lib/index.js [hunk 3]
```diff
@@ -0,0 +1 @@
...rn t[n]},set:function(e){return t[n]=e,e}})}),e}catch(r){p([r])}for(var o in t)c.call(t,o)&&(e[o]=t[o]);return e}function p(e){try{t.emit("internal-error",e)}catch(n){}}return t||(t=o),n.inPlace=f,n.flag=a,n}},{}]},{},["loader",2,16,5,3,4]);\n;NREUM.loader_config={accountID:"'+this.options.accountID+'",trustKey:"'+this.options.trustKey+'",agentID:"'+this.options.agentID+'",licenseKey:"'+this.options.license+'",applicationID:"'+this.options.applicationID+'"}\n;NREUM.info={beacon:"'+this.options.beacon+'",errorBeacon:"'+this.options.beacon+'",licenseKey:"'+this.options.license+'",applicationID:"'+this.options.applicationID+'",sa:1}',Object.keys(this.options).forEach(function(a){if(!c.options[a])throw new Error(a+' argument is required')})}return _createClass(a,[{key:'apply',value:function apply(a){var b=this;a.hooks.compilation.tap('HtmlWebpackNewRelicPlugin',function(a){_htmlWebpackPlugin2.default.getHooks(a).alterAssetTagGroups.tap('HtmlWebpackNewRelicPlugin',function(a){var c=_htmlWebpackPlugin2.default.createHtmlTagObject('script',{type:'text/javascript'},b.newrelicstring);a.headTags.push(c)})})}}]),a}();exports.default=HtmlWebpackNewRelicPlugin,module.exports=exports['default'];
```
Comment (robrap) [inline lib/index.js:1]: Where is this script coming from? It also doesn't have a final line break.

File: src/index.js [hunk 4]
```diff
@@ -37,7 +38,7 @@ window.NREUM||(NREUM={}),__nr_require=function(t,e,n){function r(n){if(!e[n]){va
             { type: 'text/javascript' },
             this.newrelicstring,
           );
-          data.bodyTags.push(newRelicScriptTag);
+          data.headTags.push(newRelicScriptTag);
```
Comment (robrap) [inline src/index.js:41]: New Relic instructions read: 
> Copy the snippet of code below and paste it as close to the top of the HEAD as possible, but after any position-sensitive META tags (X-UA-Compatible and charset).

Does the README need any special instructions for ensuring it is inserted in the right place?

**Review by robrap — COMMENTED** [review]
Providing a first pass review. Note that the version also probably needs to be updated.

File: lib/index.js (hunk 3 above)
Comment (robrap) [reply lib/index.js:1]: `lib` is in `.gitignore`, so this file should be removed I think.

File: test/index.js [hunk 5]
```diff
@@ -15,14 +15,15 @@ describe('HtmlWebpackNewRelicPlugin', () => {
     agentID: '343434',
     trustKey: '565656',
     license: '123456',
-    applicationID: '654321'
+    applicationID: '654321',
+    beacon: 'bam-cell.nr-data.net'
```
Comment (robrap) [inline test/index.js:19]: 1. What is the use case for needing to update this?
2. This isn't tested.

File: package.json (hunk 2 above)
Comment (victorrodrigues) [reply package.json:16]: @Ilya-Hubich `prepare` is part of the the [release process](https://docs.npmjs.com/cli/v7/using-npm/scripts#life-cycle-scripts). Can you revert this?

File: test/index.js (hunk 5 above)
Comment (victorrodrigues) [reply test/index.js:19]: i agree with @robrap. There is corner case that you need to use different beacon?

File: src/index.js (hunk 4 above)
Comment (victorrodrigues) [reply src/index.js:41]: As discussed in #4  we need to ensure that the script is placed not only in the head tag but, after the meta tags and before all other scripts. I'll think about it and figure this out as soon as possible.

**Comment by victorrodrigues** [comment]
Closing this due to no response from @Ilya-Hubich. Feel free to reopen it.
